wrappers/python/src/swig_doxygen/swig_lib/python/extend.i

```diff
@@ -55,26 +55,32 @@
          - getEnergy (bool=False) whether to store potential and kinetic energy in the State
          - getParameter (bool=False) whether to store context parameters in the State
          - enforcePeriodicBox (bool=False) if false, the position of each particle will be whatever position is stored in the Context, regardless of periodic boundary conditions.  If true, particle positions will be translated so the center of every molecule lies in the same periodic box.
-         - groups (int=-1) a set of bit flags for which force groups to include when computing forces and energies.  Group i will be included if (groups&(1<<i)) != 0.  The default value includes all groups.
+         - groups (set={0,1,2,...,31}) a set of indices for which force groups
+           to include when computing forces and energies. The default value
+           includes all groups. groups can also be passed as an unsigned integer
+           interpreted as a bitmask, in which case group i will be included if
+           (groups&(1<<i)) != 0.
         """
-        
-        if getPositions: getP=1
-        else: getP=0
-        if getVelocities: getV=1
-        else: getV=0
-        if getForces: getF=1
-        else: getF=0
-        if getEnergy: getE=1
-        else: getE=0
-        if getParameters: getPa=1
-        else: getPa=0
-        if enforcePeriodicBox: enforcePeriodic=1
-        else: enforcePeriodic=0
+        import functools, operator
+
+        getP, getV, getF, getE, getPa, enforcePeriodic = map(bool,
+            (getPositions, getVelocities, getForces, getEnergy, getParameters,
+             enforcePeriodicBox))
+
+        try:
+            groups = set(groups)
```

Comment:
@swails instead of using the `TypeError` thrown here if `groups` is an integer to trigger the control flow, I'll move it out into a more documented check. Obviously I'll still keep the duck typing.